### ISSUE TITLE
Metadata XML writing is intensive

### ISSUE THREAD

**Issue by PaulTalbot-INL** [idaholab/raven#52]
While writing the results of a 10k MC sampling with 671 input variables, the CSV was written successfully but writing the accompanying XML file crashed the available memory on a node on the HPC.

Perhaps we should consider more memory-efficient ways of writing out this file.  I suspect the issue is in construction the large XML tree before writing it, as we've run into this before with writing large BasicStatistics XML files.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [ ] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

**Comment by maljovec** [comment]
In addition to this, the file sizes of a 100k Monte Carlo sample set I generated did not seem to make sense.  The data consists of 54 columns in a PointSet and the csv version of the data was ~50MB. The xml was ~200MB, and the HDF5 was ~800MB.  There seems to be some inefficiencies in how we are storing data, especially in the metadata. Not to mention reading this data from HDF5 was really slow. I can avoid it by loading from csv without the xml files present. I did not attempt to load the file with the xml present as I assume that will incur a lot of overhead with the metadata.
